test/evp_extra_test.c

```diff
@@ -4647,6 +4652,74 @@ const OPTIONS *test_get_options(void)
     return options;
 }
 
+/* Test that trying to sign with a public key errors out gracefully */
+static int test_ecx_not_private_key(int tst)
+{
+    EVP_PKEY *pkey = NULL;
+
+    const unsigned char msg[] = { 0x00, 0x01, 0x02, 0x03 };
+    int testresult = 0;
+    EVP_MD_CTX *ctx = NULL;
+    unsigned char *mac = NULL;
+    size_t maclen = 0;
+    unsigned char *pubkey;
+    size_t pubkeylen;
+
+    if (testctx != NULL)
+        return TEST_skip("ecx keys not supported in null provider");
+
+    switch (keys[tst].type) {
+        case NID_X25519:
+        case NID_X448:
+            return TEST_skip("signing not supported for X25519/X448");
+    }
+
+    /* Check if this algorithm supports public keys */
+    if (keys[tst].pub == NULL)
+        return 1;
+
+    pubkey = (unsigned char *)keys[tst].pub,
```

Comment:
Please do not use comma operator unnecessarily.

Reply:
i realized this was a typo after getting confused by this comment, good eyes!